src/node_file.cc

```diff
@@ -2727,6 +2730,219 @@ static void Mkdtemp(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+static bool FileURLToPathImpl(Environment* env,
+                              const ada::url_aggregator& file_url,
+                              std::string& result_file_path) {
+  if (file_url.type != ada::scheme::FILE) {
+    env->isolate()->ThrowException(ERR_INVALID_URL_SCHEME(env->isolate()));
+
+    return false;
+  }
+
+  std::string_view pathname = file_url.get_pathname();
+#if _WIN_32
+  std::string pathname_escaped_slash;
+
+  for (size_t i = 0; (i + 2) < pathname.size(); i++) {
```

Comment:
Can you include a few comments in this section? I also suggest removing the blank lines.